tests/RouteDispatcherFactoryTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Bauhaus\HttpHandler;
+
+use FastRoute\Dispatcher;
+use PHPUnit\Framework\TestCase;
+
+class RouteDispatcherFactoryTest extends TestCase
+{
+    /**
+     * @dataProvider invalidEndpointDataprovider
+     */
+    public function testWhenGivenEndpointIsInvalidThenThrowException(string $endpoint, string $message): void
+    {
+        $config = [$endpoint => []];
+
+        $this->expectException(InvalidEndpoint::class);
+        $this->expectExceptionMessage($message);
+
+        (new RouteDispatcherFactory($config))->create();
+    }
+
+    /**
+     * @dataProvider validEndpointDataprovider
+     */
+    public function testWhenGivenEndpointIsValidThenMatchEndpoint(string $endpoint, string $method, string $uri): void
+    {
+        $config = [$endpoint => []];
+        $dispatcher = (new RouteDispatcherFactory($config))->create();
+
+        $info = $dispatcher->dispatch($method, $uri);
+
+        $this->assertEquals(Dispatcher::FOUND, $info[0]);
+    }
+
+    /**
+     * @return array<string, array<string, string>>
+     */
+    public function invalidEndpointDataprovider(): array
+    {
+        return [
+            'empty' => [
+                'endpoint' => '',
+                'message' => "The endpoint '' has an invalid format, expected format is 'METHOD URI'",
```

Comment:
I think it would be clearer if the message would be `expected format is '<METHOD> <URI>'`